src/translations.cpp

```diff
@@ -183,6 +193,46 @@ void set_language()
     reload_names();
 }
 
+#if (defined MACOSX)
+bool string_starts_with( std::string s, std::string prefix )
+{
+    return s.compare( 0, prefix.length(), prefix ) == 0;
+}
+
+std::string getOSXSystemLang()
+{
+    // Get the user's language list (in order of preference)
+    CFArrayRef langs = CFLocaleCopyPreferredLanguages();
+    if( CFArrayGetCount( langs ) == 0 ) {
+        return "en_US";
+    }
+
+    const char *lang_code_raw = CFStringGetCStringPtr(
+                                    ( CFStringRef )CFArrayGetValueAtIndex( langs, 0 ),
+                                    kCFStringEncodingUTF8 );
+    if( !lang_code_raw ) {
+        return "en_US";
+    }
+
+    // Convert to the underscore format expected by gettext
+    std::string lang_code( lang_code_raw );
+    std::replace( lang_code.begin(), lang_code.end(), '-', '_' );
+
+    /**
+     * Handle special case for simplified/traditional Chinese. Simp/trad
+     * is actually denoted by the region code in older iterations of the
+     * language codes, whereas now (at least on OS X) region is distinct.
+     * That is, CDDA expects 'zh_CN' but OS X might give 'zh-Hans-CN'.
+     */
+    if( string_starts_with( lang_code, "zh_Hans" ) ) {
+        return "zh_CN";
+    } else if( string_starts_with( lang_code, "zh_Hant" ) ) {
+        return "zh_TW";
+    }
+    return lang_code;
```

Comment:
This return value should be checked with ```isValidLanguage``` to make sure there is a translation for that language.